src/packet/signature.js

```diff
@@ -331,30 +339,22 @@ class SignaturePacket {
   }
 
   // V4 signature sub packets
-
   readSubPacket(bytes, trusted = true) {
     let mypos = 0;
 
-    const readArray = (prop, bytes) => {
-      this[prop] = [];
-
-      for (let i = 0; i < bytes.length; i++) {
-        this[prop].push(bytes[i]);
-      }
-    };
-
     // The leftmost bit denotes a "critical" packet
     const critical = bytes[mypos] & 0x80;
     const type = bytes[mypos] & 0x7F;
 
     // GPG puts the Issuer and Signature subpackets in the unhashed area.
-    // Tampering with those invalidates the signature, so we can trust them.
-    // Ignore all other unhashed subpackets.
-    if (!trusted && ![
+    // Tampering with those invalidates the signature, so we still trust them and parse them.
+    // All other unhashed subpackets are not processed further.
+    const allowedUntrustedTypes = new Set([
```

Comment:
`allowedUnhashedSubpackets`?
And, if you're going to the trouble of making a separate variable, maybe you could put it at the top-level of the file, in order not to have to create it again every time this function is called?

Reply:
I called them `Untrusted` because we use the `trusted` variable, I would either rename that as well, or leave `Untrusted`, otherwise I find the mix a bit confusing

Reply:
OK, fair enough. Yeah, I would rename that as well, because indeed "hashed" and "trusted" doesn't perfectly correspond, exactly because of this variable (so you could also call it `trustedUnhashedSubpackets`, for example).